mongodump/mongodump.go

```diff
@@ -513,6 +516,7 @@ func (dump *MongoDump) dumpIterToWriter(
 			return fmt.Errorf("error writing to file: %v", err)
 		}
 		progressCount.Inc(1)
+		dump.count += 1
```

Comment:
I would have prefered to make something like `_, dump.count = progressCount.Progress()` just below rather than incrementing two values at the same time, but the `Progress` method isn't accessible with a `progress.Updateable`. I actually wonder: why is there is two different `progress` interfaces @3rf ?